File: Frontend Code/src/services/analyticsService.js
import api from "./api";

const analyticsService = {
  getSystemAnalytics: async () => {
    const { data } = await api.get("/analytics/system");
    return data;
  },

  getInstitutionAnalytics: async () => {
    const { data } = await api.get("/analytics/institution");
    return data;
  },

  getLabAnalytics: async () => {
    const { data } = await api.get("/analytics/lab");
    return data;
  },

  getRevenueByEquipment: async () => {
    const { data } = await api.get("/analytics/revenue/equipment");
    return data;
  },

  getRevenueByLab: async () => {
    const { data } = await api.get("/analytics/revenue/lab");
    return data;
  },

  getEquipmentUsage: async () => {
    const { data } = await api.get("/analytics/equipment/usage");
    return data;
  },

  getTopEquipment: async () => {
    const { data } = await api.get("/analytics/equipment/top");
    return data;
  },

  getLeastEquipment: async () => {
    const { data } = await api.get("/analytics/equipment/least");
    return data;
  },

  getMonthlyBookings: async () => {
    const { data } = await api.get("/analytics/bookings/monthly");
    return data;
  },

  getBookingTrend: async () => {
    const { data } = await api.get("/analytics/bookings/trend");
    return data;
  },

  getWaitingQueueAnalytics: async () => {
    const { data } = await api.get("/analytics/waiting-queue");
    return data;
  },
};

export default analyticsService;

File: src/main/java/com/LabResourceUtilizationPlatform/Service/AnalyticsService.java
package com.LabResourceUtilizationPlatform.Service;

import com.LabResourceUtilizationPlatform.Dtos.Response.*;

import java.util.List;

public interface AnalyticsService {

    // Dashboard Summary
    AnalyticsDashboardResponse getSystemAnalytics();

    AnalyticsDashboardResponse getInstitutionAnalytics();

    AnalyticsDashboardResponse getLabAnalytics();

    // Revenue
    List<RevenueResponse> getRevenueByEquipment();

    List<RevenueResponse> getRevenueByLab();

    // Equipment Usage
    List<EquipmentUsageResponse> getEquipmentUsage();

    List<TopEquipmentResponse> getTopUsedEquipment();

    List<TopEquipmentResponse> getLeastUsedEquipment();

    // Booking Analytics
    List<MonthlyBookingResponse> getMonthlyBookings();

    List<BookingTrendResponse> getBookingTrend();

    // Waiting Queue
    List<WaitingQueueAnalyticsResponse> getWaitingQueueAnalytics();
}